modules/programs/zsh.nix

```diff
@@ -79,6 +100,18 @@ in
 
       home.file.".zshrc".text = ''
         ${export "HISTSIZE" cfg.historySize}
+        ${export "HISTFILE" cfg.historyFile}
+
+        setopt HIST_FCNTL_LOCK
+        ${if cfg.historyIgnoreDups then "setopt" else "unsetopt"} HIST_IGNORE_DUPS
+        ${if cfg.historyShare then "setopt" else "unsetopt"} SHARE_HISTORY
+
+        # Tell zsh how to find installed completions
+        for p in ''${(z)NIX_PROFILES}; do
+          fpath+=($p/share/zsh/site-functions $p/share/zsh/$ZSH_VERSION/functions)
+        done
+
+        HELPDIR="${pkgs.zsh}/share/zsh/$ZSH_VERSION/help"
```

Comment:
I'll just merge this as-is. We can remove this workaround when/if the zsh package in NixOS is changed.